### ISSUE TITLE
Hematovision 

### ISSUE THREAD

**Issue by naginenichandrika** [naginenichandrika/Hematovision-advanced-blood-cells-classification-using-transfer-learning-#1]
Hematovision 
# hemito_vision.rb
require 'sinatra'
require 'fileutils'

set :public_folder, 'public'
set :upload_folder, 'uploads'

# Ensure folders exist
FileUtils.mkdir_p(settings.public_folder)
FileUtils.mkdir_p(settings.upload_folder)

get '/' do
  <<-HTML
  <!DOCTYPE html>
  <html>
  <head>
    <title>Hemito Vision</title>
    <style>
      body {
        font-family: Arial, sans-serif;
        background-color: #f4f6f8;
        display: flex;
        justify-content: center;
        align-items: center;
        height: 100vh;
      }
      .container {
        background: white;
        padding: 30px;
        border-radius: 10px;
        box-shadow: 0 0 10px rgba(0,0,0,0.1);
        text-align: center;
      }
      input[type="file"] {
        margin: 20px 0;
      }
      input[type="submit"] {
        background-color: #3498db;
        color: white;
        border: none;
        padding: 10px 20px;
        border-radius: 5px;
        cursor: pointer;
      }
    </style>
  </head>
  <body>
    <div class="container">
      <h1>Hemito Vision</h1>
      <p>Upload a blood smear image for classification</p>
      <form action="/predict" method="POST" enctype="multipart/form-data">
        <input type="file" name="file" accept="image/*" required><br>
        <input type="submit" value="Predict">
      </form>
    </div>
  </body>
  </html>
  HTML
end

post '/predict' do
  unless params[:file]
    return "No file uploaded."
  end

  filename = params[:file][:filename]
  file = params[:file][:tempfile]

  filepath = File.join(settings.upload_folder, filename)
  File.open(filepath, 'wb') { |f| f.write(file.read) }

  # Call Python prediction
  result = `python3 -c "
import tensorflow as tf
from tensorflow.keras.models import load_model
from tensorflow.keras.preprocessing import image
import numpy as np
model = load_model('hemito_vision_blood_cell_classifier.h5')
img = image.load_img('#{filepath}', target_size=(224, 224))
x = image.img_to_array(img)/255.0
x = x.reshape(1, 224, 224, 3)
pred = model.predict(x)
classes = ['RBC', 'WBC', 'Platelet', 'Blast Cell']
idx = int(np.argmax(pred))
conf = round(100 * float(np.max(pred)), 2)
print(classes[idx] + '|' + str(conf))
"`

  label, confidence = result.strip.split('|')

  <<-HTML
  <!DOCTYPE html>
  <html>
  <head>
    <title>Prediction Result</title>
    <style>
      body {
        font-family: Arial, sans-serif;
        background-color: #f4f6f8;
        display: flex;
        justify-content: center;
        align-items: center;
        height: 100vh;
      }
      .container {
        background: white;
        padding: 30px;
        border-radius: 10px;
        box-shadow: 0 0 10px rgba(0,0,0,0.1);
        text-align: center;
      }
      a {
        text-decoration: none;
        color: #3498db;
        display: block;
        margin-top: 20px;
      }
    </style>
  </head>
  <body>
    <div class="container">
      <h1>Prediction Result</h1>
      <p><strong>Class:</strong> #{label}</p>
      <p><strong>Confidence:</strong> #{confidence}%</p>
      <a href="/">Try Another</a>
    </div>
  </body>
  </html>
  HTML
end
